### ISSUE TITLE
Bug: Return and Yield overly unwraps the return type (DOC404 false positive)

### ISSUE THREAD

**Issue by pseudo-rnd-thoughts** [jsh9/pydoclint#288]
### Summary
When a function has **both** a `yield` statement and a `return` statement, and its return annotation is `AsyncIterator[Dict[str, Any]]` (or similar), pydoclint extracts the yield type **twice**, produces `(str, Any)` (not `Dict[str, Any]` as expected). This causes a DOC404 false positive against a correctly-typed docstring.

### Version

- pydoclint 0.8.3
- Python 3.12

### Minimal reproducer

```python
from typing import AsyncIterator, Dict, Any

async def f(self, batch: Dict[str, Any]) -> AsyncIterator[Dict[str, Any]]:
    """Do work.

    Args:
        batch: The input batch.

    Yields:
        Dict[str, Any]: The outputs.
    """
    if not batch:
        yield {}
        return            # <-- presence of `return` triggers the bug
    yield {"a": 1}
```
Output
```
repro.py
    5: DOC404: Method `f` yield type(s) in docstring not consistent with
    the return annotation. The yield type (the 0th arg in Generator[...]
    /Iterator[...]): (str, Any); docstring "yields" section types:
    Dict[str, Any]
```

Context: https://github.com/ray-project/ray/blob/d6627fe17c26dc33fe4a0d4503ed73874e6575df/python/ray/llm/_internal/batch/stages/base.py#L106 where I was fixing the docstrings (https://github.com/ray-project/ray/pull/63632)

### Root cause
Getting claude to diagnosise the issue, it claims 

> in `pydoclint/visitor.py::Visitor.checkReturnAndYield` (around lines 950–967 in 0.8.3) `extract` correctly returns `"Dict[str, Any]"` for `AsyncIterator[Dict[str, Any]]`. But that already-extracted value is then wrapped as a `ReturnAnnotation` and passed into `checkYieldTypesForViolations`, which **runs the same extractor again** (`visitor_helper.py:543-548`) with `hasIteratorOrIterableAsReturnAnnotation=True` still set. The second pass parses `"Dict[str, Any]"`, takes `.slice`, and unparses the `Tuple(str, Any)` node, producing `(str, Any)`. The normal yields-only path (`checkYields`) calls the extractor exactly once and works correctly — this bug is specific to functions that go through `checkReturnAndYield` (i.e. have both `yield` and `return`).
> To fix it, the `checkReturnAndYield`, pass the **original** `returnAnno` into `checkYieldTypesForViolations` (which will extract once internally), instead of pre-extracting and then re-passing. Either:
> - Drop the outer `extract(...)` call and let `checkYieldTypesForViolations` do the extraction, or
> - Have `checkYieldTypesForViolations` accept an already-extracted yield type and skip its internal extraction.

I've tested claude's hypothesis and taking the first option works with no other problems

**Comment by jsh9** [comment]
Hi @pseudo-rnd-thoughts , thank you very much for raising this issue.

@StressTestor is kind enough to make a code change in #291 . I merged it and it will be published as the next version 0.8.7 in a few minutes.
